packages/modal/src/components/drawer/Drawer.tsx

```diff
@@ -16,31 +40,27 @@ export interface DrawerProps
     | "style"
     | "parentSelector"
   > {
-  width?: string;
   background?: string;
   zIndex?: number;
   onRequestClose?: () => void;
-  /**
-   * Which direction the drawer will appear from.
-   * @default left
-   * @param {String('left'|'right')}
-   */
-  slideFrom?: SlideFrom;
   /**
    * Portal target, HTML element. If not set, portal is not used.
    */
   portalTarget?: HTMLElement | null;
 }
 
 export const Drawer: React.FC<DrawerProps> = ({
-  width = "370px",
   background,
   zIndex,
   children,
   slideFrom = "left",
   portalTarget,
   ...reactModalProps
 }) => {
+  const height =
+    "height" in reactModalProps ? reactModalProps.height : undefined;
+  const width = "width" in reactModalProps ? reactModalProps.width : undefined;
```

Comment:
Maybe something like this:

```suggestion
  ...drawerProps
}) => {
  const {
    width,
    height,
    ...reactModalProps
  } = drawerProps as DrawerBaseLeftRight & DrawerBaseTopBottom;
```

Reply:
That is not type-safe, if prop `width` changes from optional to required, then variable `width` will be typed as `string` but in reality it is `string | undefined`.

This is minor, obviously, but the original code is safer.